src/Extensions.hs

```diff
@@ -63,30 +64,27 @@ getPackageExtentions cabalFile = do
   where
     perModuleParseMerge :: FilePath -> [OnOffExtension] -> IO ExtensionsResult
     perModuleParseMerge path cabalExts = do
-        -- TODO: catch and return SourceNotFound
         moduleRes <- parseFile path
         pure $ mergeCabalAndModule cabalExts path moduleRes
 
 {- | By given path to @.cabal@ file and 'Hashmap' of sources of all Haskell
 modules, analyse extensions for each Haskell module and return the corresponding
 'HashMap'.
-
-__Throws__:
-
-* 'Extensions.Cabal.CabalException'
 -}
 getPackageExtentionsBySources
     :: FilePath  -- ^ Path to @.cabal@ file.
     -> Map FilePath ByteString  -- ^ Path to modules with corresponding sources.
     -> IO (Map FilePath ExtensionsResult)
 getPackageExtentionsBySources cabalFile sourcesMap = do
-    cabalMap <- parseCabalFileExtensions cabalFile
-    pure $ merge
-        (mapMissing cabalNotSource) -- in cabal but not in sources
-        (mapMissing sourceNotCabal) -- in sources but not in cabal
-        (zipWithMatched cabalAndSource) -- in cabal and sources
-        cabalMap
-        sourcesMap
+    cabalRes <- parseCabalHandleException cabalFile
+    pure $ case cabalRes of
+        Left err -> Map.map (const err) sourcesMap
```

Comment:
Actually, you can write `err <$ sourcesMap` since `Map` implements a `Functor` instance. And remember reading something about efficient implementation of a function, that sets all values to some specific value via `<$`. From the implementation I can see that the implementation is manual for this function:

* https://hackage.haskell.org/package/containers-0.6.2.1/docs/src/Data.Map.Internal.html#line-4196